homeassistant/components/isy994/climate.py

```diff
@@ -0,0 +1,360 @@
+from homeassistant.util.temperature import convert as convert_temperature
+
+_LOGGER = logging.getLogger(__name__)
+
+ISY_CURRENT_TEMPERATURE = 'ST'
+ISY_CURRENT_HUMIDITY = 'CLIHUM'
+ISY_TARGET_TEMP_HIGH = 'CLISPC'
+ISY_TARGET_TEMP_LOW = 'CLISPH'
+ISY_OPERATION_MODE = 'CLIMD'
+ISY_OPERATION_STATE = 'CLIHCS'
+ISY_FAN_MODE = 'CLIFS'
+ISY_UOM = 'UOM'
+
+DEFAULT_MIN_TEMP = 10
+DEFAULT_MAX_TEMP = 30
+
+# ISY Operation Modes (as passed back from ISY). Since these don't match
+# the required HASS Climate States, they will be split into HASS States
+# and Hold Modes in a separate function. Kept here for debugging purposes.
+ISY_MODE_OFF = 'off'
+ISY_MODE_HEAT = 'heat'
+ISY_MODE_COOL = 'cool'
+ISY_MODE_AUTO = 'auto'
+ISY_MODE_FAN_ONLY = 'fan_only'
+ISY_MODE_PROG_HEAT = 'program_heat'
+ISY_MODE_PROG_COOL = 'program_cool'
+ISY_MODE_PROG_AUTO = 'program_auto'
+
+# Translate ISY UOM 98
+VALUE_TO_ISY_MODE = {
+    0: ISY_MODE_OFF,
+    1: ISY_MODE_HEAT,
+    2: ISY_MODE_COOL,
+    3: ISY_MODE_AUTO,
+    4: ISY_MODE_FAN_ONLY,
+    5: ISY_MODE_PROG_AUTO,
+    6: ISY_MODE_PROG_HEAT,
+    7: ISY_MODE_PROG_COOL
+}
+
+MODE_TO_VALUE = {}
+for key in VALUE_TO_ISY_MODE:
+    MODE_TO_VALUE[VALUE_TO_ISY_MODE[key]] = key
+
+# Translate ISY Operation Mode to HASS States & Hold Modes
+VALUE_TO_HASS_MODE = {
+    0: (STATE_OFF, 'temp'),
+    1: (STATE_HEAT, 'temp'),
+    2: (STATE_COOL, 'temp'),
+    3: (STATE_AUTO, 'temp'),
+    4: (STATE_FAN_ONLY, 'temp'),
+    5: (STATE_AUTO, 'program'),
+    6: (STATE_HEAT, 'program'),
+    7: (STATE_COOL, 'program')
+}
+
+# Translate ISY UOM 99
+FAN_VALUE_TO_MODE = {
+    7: STATE_ON,
+    8: STATE_AUTO
+}
+
+FAN_MODE_TO_VALUE = {}
+for key in FAN_VALUE_TO_MODE:
+    FAN_MODE_TO_VALUE[FAN_VALUE_TO_MODE[key]] = key
+
+# Translate ISY UOM 66
+VALUE_TO_STATE = {
+    0: STATE_IDLE,
+    1: STATE_HEAT,
+    2: STATE_COOL,
+    3: STATE_FAN_ONLY
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the ISY994 thermostat platform."""
+    devices = []
+
+    for node in hass.data[ISY994_NODES][DOMAIN]:
+        _LOGGER.debug('Adding ISY node %s to Climate platform', node)
+        devices.append(ISYThermostatDevice(node))
+
+    async_add_entities(devices)
+
+
+class ISYThermostatDevice(ISYDevice, ClimateDevice):
+    """Representation of an ISY994 thermostat device."""
+
+    def __init__(self, node) -> None:
+        """Initialize the ISY Thermostat Device."""
+        super().__init__(node)
+        self._node = node
+        self._uom = ''
+        if len(self._node.uom) == 1:
+            self._uom = self._node.uom[0]
+        self._current_status = None
+        self._current_hold_mode = None
+        self._current_humidity = 0
+        self._target_temp_low = 0
+        self._target_temp_high = 0
+        self._fan_mode = None
+        self._current_operation = None
+        self._current_isy_operation = None
```

Comment:
This looks like it's only assigned, not used.

Reply:
It's set to the mode that is reported back from the ISY in Line 142 which I am splitting out into `operation_mode` and `hold_mode` to meet the climate modules' requirements for mode.  It was left in there for debugging and as a placeholder depending on the direction home-assistant/architecture#22 goes.

Reply:
Unused attributes should be removed before merge.